lib/stdlib/src/erl_scan.erl

```diff
@@ -828,6 +857,218 @@ scan_char([], St, Line, Col, Toks) ->
+            if
+                Qs =/= 0, ?WHITE_SPACE(C) ->
+                    %% White space while searching for end quote chars
+                    if
+                        Qs =:= Tqs#tqs.qs ->
+                            %% White space before first end quote char
+                            %% - just collect
+                            scan_tqstring_line(
+                              Ncs, St, Line, Ncol, Toks,
+                              Qs, Tqs, Nacc);
+                        true ->
+                            %% White space after too few quote chars
+                            %% - stop searching for end quote chars
+                            scan_tqstring_line(
+                              Ncs, St, Line, Ncol, Toks,
+                              0, Tqs, Nacc)
+                    end;
+                ?UNI255(C) ->
+                    scan_tqstring_line(
+                      Ncs, St, Line, Ncol, Toks,
+                      0, Tqs, Nacc); % Stop searching for end quote chars
+                ?CHAR(C) ->
+                    %% Illegal Unicode character
+                    scan_error(
+                      {illegal,character}, Line, Col, Line, Ncol, Ncs)
+            end;
+        [] ->
+            {more,
+             {[], St, Col, Toks, Line, {Qs, Tqs, Acc},
+              fun scan_tqstring_line/6}};
+        eof ->
+            ContentR = [Acc|Tqs#tqs.content_r],
+            Estr = string:slice(tqstring_chars_r(ContentR), 0, 16),
+            scan_error(
+              {string,{$",Tqs#tqs.qs},Estr},%"
+              Tqs#tqs.line, Tqs#tqs.col, Line, Col, eof)
+    end.
+
+%% Strip last line newline,
+%% strip indentation,
+%% check white space on first line,
+%% create the string token,
+%% done
+%%
+scan_tqstring_finish(Cs, St, Line, Col, Toks, Tqs, IndentR) ->
+    %% IndentR :: Indentation characters, reversed
+    #tqs{ line = Line0, col = Col0, content_r = ContentR } = Tqs,
+    NcontentR = strip_last_line_newline_r(ContentR),
+    case
+        tqstring_finish(lists:reverse(IndentR), NcontentR, Line-1)
+    of
+        Content when is_list(Content) ->
+            Qs = Tqs#tqs.qs,
+            Anno =
+                anno(
+                  Line0, Col0, St,
+                  ?STR(
+                     string, St, Chars,
+                     lists_duplicate(
+                       Qs, $",%"
+                       tqstring_chars_r(
+                         ContentR, lists:duplicate(Qs, $"))))),%"
+            scan1(Cs, St, Line, Col, [{string,Anno,Content}|Toks]);
+        {Tag=indentation, ErrorLine, ErrorCol} ->
+            scan_error(
+              Tag, ErrorLine, new_column(Col, ErrorCol),
+              Line, Col, Cs);
+        {Tag=white_space, N} ->
+            scan_error(
+              Tag, Line0, incr_column(Col0, Tqs#tqs.qs+N),
+              Line, Col, Cs)
+    end.
+
+%% Reconstruct the scanned triple-quoted string from content lines
+%%
+tqstring_chars_r(ContentR) ->
```

Comment:
Is `tqstring_chars_s` the same as:
```erl
lists:foldl(fun (X, Acc) -> [lists:reverse(X)|Acc] end, [], ContentR).
```
Just checking whether I understood it and wondering if it makes more sense the shorter version or this recursive version

Reply:
Almost.  Changing to `lists:reverse(X, Acc)` makes it equivalent.  Which leads to this beauty, that I will change to, thanks for the suggestion:
``` erlang
tqstring_chars_r(ContentR, String) ->
    lists:foldl(fun lists:reverse/2, String, ContentR).
```